spec/src/main/asciidoc/cdi.asciidoc

```diff
@@ -19,4 +19,156 @@
 
 If a CDI implementation is available, instances of `ManagedExecutor` and `ThreadContext` can be injected as CDI beans.
 
-TODO write the documentation after discussion on Config annotations is resolved
\ No newline at end of file
+=== Injecting `ManagedExecutor`
+
+The simplest use case is a new executor instance per injection point, where the configuration is equivalent to the default values of the `@ManagedExecutorConfig` annotation.
+
+[source, java]
+----
+@Inject
+ManagedExecutor executor;
+----
+
+==== Configuring `ManagedExecutor`
+
+Oftentimes there is a need to configure the executor instance to limit the number of asynchronous tasks that it can start or define which contexts to propagate.
+The `@ManagedExecutorConfig` annotation allows to configure all aspects of an injected `ManagedExecutor` instance.
+Note that using this annotation only, new instance of `ManagedExecutor` is created for every injection point even if those injection points have equivalent configuration.
```

Comment:
Minor grammar updates:
allows to configure --> allows you to configure
Note that using --> Note that when using
new instance of --> a new instance of